build-system/tasks/helpers.js

```diff
@@ -681,11 +635,28 @@ async function minifyWithTerser(destDir, destFilename, options) {
  */
 async function compileJs(srcDir, srcFilename, destDir, options) {
   options = options || {};
-  if (options.minify) {
-    return compileMinifiedJs(srcDir, srcFilename, destDir, options);
-  } else {
-    return compileUnminifiedJs(srcDir, srcFilename, destDir, options);
+
+  async function compile(modifiedFile) {
```

Comment:
maintainability nit: This function name can be easily confused with `compile()` in `compile.js` (the inner closure compiler implementation).

Since it implements the innards of `compileJs()`, maybe call it `doCompileJs()`?

Reply:
Done